### ekf_filter_node config cho MangoMobileRobot ###
ekf_filter_node:
  ros__parameters:
    
    frequency: 30.0
    two_d_mode: true
    publish_tf: true

    map_frame: map              
    odom_frame: odom            
    base_link_frame: base_footprint 
    world_frame: odom           

    # -----------------------------------------------
    # CẤU HÌNH ĐẦU VÀO
    # -----------------------------------------------
    
    # SỬA LẠI Ở ĐÂY: Trỏ thẳng vào topic đã được Node Python vá lỗi frame
    odom0: /odom/fixed

    # Ma trận cấu hình (X, Y, Yaw, vX, vYaw)
    odom0_config: [true,  true,  false,
                   false, false, true,
                   true,  false, false,
                   false, false, true,
                   false, false, false]

    odom0_differential: false
    odom0_relative: false
    odom0_queue_size: 10